tests_e2e/test_selenium_trello.py

```diff
@@ -0,0 +1,58 @@
+import os
+from threading import Thread
+from flask import Flask, render_template, request, redirect, url_for
+import requests
+import unittest
+from unittest.mock import patch
+import pytest
+import pytest_html
+from selenium import webdriver
+import app
+import config as cf
+
+@pytest.fixture(scope='module')
+def test_app():
+    # Create the new board & update the board id environment variable
+    board_id = create_trello_board('Testing')
+    os.environ['TRELLO_BOARD_ID'] = board_id
+    # construct the new application
+    application = app.create_app()
+    # start the app in its own thread.
+    thread = Thread(target=lambda: application.run(use_reloader=False))
+    thread.daemon = True
+    thread.start()
+    yield app
+    # Tear Down
+    thread.join(1)
+    delete_trello_board(board_id)
+
+#Fixture for Firefox
+@pytest.fixture(scope="module")
+def driver(request):
+    ff_driver = webdriver.Firefox()
+    request.cls.driver = ff_driver
+    yield
+    ff_driver.close()
```

Comment:
The exercise pdf gives you the necessary code for the "driver" fixture but you've written something else here and it doesn't work. Did you try running your selenium test?